core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -150,21 +152,12 @@ private void checkDiskSpace(DockerClient client) {
             callback.awaitCompletion();
             String logResults = callback.toString();
 
-            int availableKB = 0;
-            int use = 0;
-            String[] lines = logResults.split("\n");
-            for (String line : lines) {
-                String[] fields = line.split("\\s+");
-                if (fields[5].equals("/")) {
-                    availableKB = Integer.valueOf(fields[3]);
-                    use = Integer.valueOf(fields[4].replace("%", ""));
-                }
-            }
-            int availableMB = availableKB / 1024;
-
-            LOGGER.info("Disk utilization in Docker environment is {}% ({} MB available)", use, availableMB);
+            DiskSpaceUsage df = parseAvailableDiskSpace(logResults);
+            LOGGER.info("Disk utilization in Docker environment is {} ({} )", 
+                    df.usedPercent.map(x -> x.toString() + "%").orElse("unknown"), 
+                    df.availableMB.map(x -> x.toString() + " MB available").orElse("unknown available"));
 
-            if (availableMB < 2048) {
+            if (df.availableMB.orElseThrow(NotAbleToGetDiskSpaceUsageException::new) < 2048) {
```

Comment:
Will this throw an exception if free space can't be found?

I think that would be too harsh - maybe the nicest behaviour if the `df` check returns nothing is to just let the process carry on. It would probably be annoying for a user to be blocked completely just because `df` doesn't work in their environment.